src/JetUserData.cc

```diff
@@ -250,7 +250,7 @@ void JetUserData::produce( edm::Event& iEvent, const edm::EventSetup& iSetup) {
       }
       // ... and gaussian smear the rest
       if (!isGenMatched && JERSF>1) {
-	double sigma = std::sqrt(JERSF * JERSF - 1) * PtResolution;
+	double sigma = std::sqrt(JERSF * JERSF - 1) * PtResolution / jet.p4();
```

Comment:
You are right, it's not jet.p4()!
It should be jet.pt(), I make a correction similar to the other line you show.